libdevice/CMakeLists.txt

```diff
@@ -0,0 +1,159 @@
+# Utility project providing various functionalities for SPIR-V devices
+# without native support of these functionalities.
+
+if (WIN32)
+  set(binary_dir "${CMAKE_RUNTIME_OUTPUT_DIRECTORY}")
+  set(install_dest bin)
+else()
+  set(binary_dir "${CMAKE_LIBRARY_OUTPUT_DIRECTORY}")
+  set(install_dest lib)
+endif()
+
+set(clang $<TARGET_FILE:clang>)
+
+if (sycl IN_LIST LLVM_ENABLE_PROJECTS)
```

Comment:
It looks a little bit strange that you build libdevice only along with sycl while saying in the commit message that want to reuse for other projects.

Reply:
This change-set just moves libdevice to the toplevel.

Reply:
"sycl" project here means sycl runtime library which is not needed in order to build devicelib, so I suggest removing requirement for "sycl" to be in LLVM_ENABLE_PROJECTS

Reply:
You are right.  I need a knob to disable libdevice build in repos that do not have SYCL support in clang, e.g. llorg forks.

I can check for an explicit knob, and set it in configure.py. What would be your recommendation?

Reply:
Ok, understand. Do not know a good way how to make sure clang has a SYCL support here.
I think the current way is error prone as it just silently does nothing if sycl is not in the list of
enabled components.

Is it possible to just not enable the libdevice component when configuring cmake in llorg forks?

Suggest rewriting to:

if (sycl NOT IN_LIST LLVM_ENABLE_PROJECTS)
  message(FATAL "....")

and then do what is currently done in the if body.

> I can check for an explicit knob, and set it in configure.py
Not sure what it means.

Reply:
Vlad, I cannot do what you suggest, since my llorg fork will not have 'sycl' in LLVM_ENABLE_PROJECTS.

I suggest adding a CMake control, e.g. LIBDEVICE_BUILD_FOR_SYCL, set it in configure.py and check it here.  Unfortunately, every developer will have to set it as well for local builds.